src/software/backend/input/network/networking/proto_multicast_listener.cpp

```diff
@@ -62,12 +64,17 @@ void SSLVisionClient::handleDataReception(const boost::system::error_code& error
         // Start listening again to receive the next data
         socket_.async_receive_from(
             boost::asio::buffer(raw_received_data_, max_buffer_length), sender_endpoint_,
-            boost::bind(&SSLVisionClient::handleDataReception, this,
+            boost::bind(&ProtoMulticastListener::handleDataReception, this,
                         boost::asio::placeholders::error,
                         boost::asio::placeholders::bytes_transferred));
 
         LOG(WARNING)
-            << "An unknown network error occurred when attempting to receive SSL Vision Data. The boost system error code is "
+            << "An unknown network error occurred when attempting to receive ReceiveProto Data. The boost system error code is "
             << error << std::endl;
     }
 }
+
+// Place all templated initializations here
+// See https://isocpp.org/wiki/faq/templates#separate-template-class-defn-from-decl
+template class ProtoMulticastListener<Referee>;
+template class ProtoMulticastListener<SSL_WrapperPacket>;
```

Comment:
Another option here is to have a separate `.tpp` file that you include at the bottom of the `.h` file. We've already used that pattern in a few place(s), like https://github.com/UBC-Thunderbots/Software/blob/master/src/software/optimization/gradient_descent_optimizer.h. 

Personally I prefer the `.tpp` because this way `ProtoMulticastListener` doesn't need to directly depend on everything it could be templated as.

Reply:
do you mean including these lines in a .tpp file and including the .tpp file at the bottom of the .h file?